eureka/lib/util.py

```diff
@@ -19,12 +19,22 @@ def readfiles(meta):
         The metadata object with added segment_list containing the sorted
         data fits files.
     """
-    meta.segment_list = []
+    if suffix is None:
+        suffix = meta.suffix
+        return_meta=True
+    else:
+        return_meta=False
```

Comment:
This is a really confusing way of changing what gets returned from this function. You should just change your code to accept the way this function was already setup and revert the changes to this function